### PR TITLE
Cache penultimate machine as well as last machine

### PR DESCRIPTION
**Pull request by joshuacolvin0**

### CODE REVIEW

File: packages/arb-avm-cpp/data_storage/src/combinedmachinecache.cpp [hunk 1]
```diff
@@ -21,11 +21,16 @@ void CombinedMachineCache::checkLastMachine(uint256_t& arb_gas_used) {
...
+    if (last_last_machine != nullptr &&
+        last_last_machine->machine_state.output.arb_gas_used < arb_gas_used) {
+        last_last_machine = nullptr;
+    }
 }
 
 void CombinedMachineCache::lastAdd(std::unique_ptr<Machine> machine) {
     std::unique_lock lock(mutex);
 
+    last_last_machine = std::move(last_machine);
```
Comment (hkalodner) [inline packages/arb-avm-cpp/data_storage/src/combinedmachinecache.cpp:33]: I think we should only do this if `last_machine` is non-empty

**Review by hkalodner — DISMISSED** [review]
LGTM

File: packages/arb-avm-cpp/data_storage/src/combinedmachinecache.cpp (hunk 1 above)
Comment (joshuacolvin0) [reply packages/arb-avm-cpp/data_storage/src/combinedmachinecache.cpp]: :+1:

**Review by hkalodner — DISMISSED** [review]
LGTM

**Review by hkalodner — APPROVED** [review]
LGTM

**Pull request merged**